core/model/modx/processors/browser/file/rename.class.php

```diff
@@ -40,10 +40,20 @@ public function process() {
         }
 
         $oldFile = $this->getProperty('path');
-        $oldFile = preg_replace('/[\.]{2,}/', '', htmlspecialchars($oldFile));
-        $name = $this->getProperty('name');
-        $name = preg_replace('/[\.]{2,}/', '', htmlspecialchars($name));
-        $success = $this->source->renameObject($oldFile, $name);
+        $oldlocale = setlocale(LC_ALL, 0);
+        setlocale(LC_ALL,'C.UTF-8');
```

Comment:
Please check here, wether setlocale is successful.

Reply:
This trouble solving if MODX Setting "locale" is correct encoding for current server environment.
Now I don't run setlocale manually, just log about encoding if function pathinfo return not what was expected